RecoParticleFlow/PFClusterProducer/src/PFClusterEMEnergyCorrector.cc

```diff
@@ -335,18 +335,33 @@ void PFClusterEMEnergyCorrector::correctEnergies(const edm::Event &evt,
     int coridx = 0;
     int regind = 0;
     if (!iseb) regind = 4;
-    if (clusFlag==1)
+    
+    ///////////////////////////////////////////////////////////////////////////////////
+    ///a hit can be ZS or forced ZS. A hit can be in Full readout or Forced to be FULL readout
+    ///if it is ZS, then clusFlag (in binary) = 0001
+    ///if it is forced ZS, then clusFlag (in binary) = 0101
+    ///if it is FR, then clusFlag (in binary) = 0011
+    ///if it is forced FR, then clusFlag (in binary) = 0111
+    ///i.e 3rd bit is set.
+    ///Even if it is forced, we should mark it is as ZS or FR. To take care of it, just check the LSB and second LSB(SLSB)
+    ///////////////////////////////////////////////////////////////////////////////////
+    int ZS_bit = clusFlag>>0&1;
+    int FR_bit = clusFlag>>1&1;
+    
+    
+
+    if (ZS_bit && !FR_bit) ///it is clusFlag==1, 5
```

Comment:
in general, yes, ints should not be negated or used like this if they are meant to be actually bits.
The way how these bits are defined (with ``` a& 1```) this does not matter.
@jainshilpi please update anyways.